cpp/src/reductions/reductions.cpp

```diff
@@ -176,7 +181,24 @@ std::unique_ptr<scalar> reduce(column_view const& col,
                                rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
-  return detail::reduce(col, agg, output_dtype, cudf::default_stream_value, mr);
+  return detail::reduce(col, agg, output_dtype, std::nullopt, cudf::default_stream_value, mr);
 }
 
+std::unique_ptr<scalar> reduce(column_view const& col,
+                               std::unique_ptr<reduce_aggregation> const& agg,
+                               data_type output_dtype,
+                               scalar const& init,
+                               rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(col.type() == init.type(), "column and initial value must be the same type");
+  if (!(agg->kind == aggregation::SUM || agg->kind == aggregation::PRODUCT ||
```

Comment:
Add this check/fail logic to the [segmented reduction here](https://github.com/rapidsai/cudf/blob/872915a5a83a23a14992b9c503bd82222031c491/cpp/src/reductions/segmented_reductions.cpp#L145) as well -- we don't support any other aggregations besides these six at the moment, but we should guard that code to prevent future problems if and when more aggregations are added.